src/cattr/gen.py

```diff
@@ -162,11 +164,18 @@ def make_dict_structure_fn(
         # For each attribute, we try resolving the type here and now.
         # If a type is manually overwritten, this function should be
         # regenerated.
-        if type is not None:
+        if converter._prefer_attrib_converters and a.converter is not None:
+            # The attribute has defined its own conversion, so pass
+            # the original value through without invoking cattr hooks
+            handler = _passthru
+        elif type is not None:
             handler = converter._structure_func.dispatch(type)
         else:
             handler = converter.structure
 
+        if not converter._prefer_attrib_converters and a.converter is not None:
+            handler = _fallback_to_passthru(handler)
```

Comment:
I don't think we need to use `_fallback_to_passthru`, we can know here and now if we will need to passthrough, right? If the handler is equal to `converter._structure_default`, the type is not `is_generic` and calling `_structure_default` raises an exception. It's a little dirty but should work.

Ideally the `structure_default` handler can be refactored to only contain the error logic, but that might be a different task.

Reply:
Sorry, I don't quite follow what you're proposing the code should do differently here.